src/databricks/labs/ucx/hive_metastore/pipelines_migrate.py

```diff
@@ -0,0 +1,142 @@
+
+        tasks = []
+        for pipeline in pipelines_to_migrate:
+            if pipeline.pipeline_id in self._skip_pipelines:
+                logger.info(f"Skipping pipeline {pipeline.pipeline_id}")
+                continue
+            tasks.append(partial(self._migrate_pipeline, pipeline))
+        if not tasks:
+            return []
+        Threads.strict("migrate pipelines", tasks)
+        return tasks
+
+    def _migrate_pipeline(self, pipeline: PipelineInfo) -> dict | list | BinaryIO | bool:
+        try:
+            return self._clone_pipeline(pipeline)
+        except DatabricksError as e:
+            if "Cloning from Hive Metastore to Unity Catalog is currently not supported" in str(e):
+                logger.error(f"{e}: Please contact Databricks to enable DLT HMS to UC migration API on this workspace")
+                return False
+            logger.error(f"Failed to migrate pipeline {pipeline.pipeline_id}: {e}")
+            return False
+
+    def _clone_pipeline(self, pipeline: PipelineInfo) -> dict | list | BinaryIO:
+        # Need to get the pipeline again to get the libraries
+        # else updating name fails with libraries not provided error
+        get_pipeline = self._ws.pipelines.get(pipeline.pipeline_id)
+        if get_pipeline.spec:
+            if get_pipeline.spec.catalog:
+                # Skip if the pipeline is already migrated to UC
+                logger.info(f"Pipeline {pipeline.pipeline_id} is already migrated to UC")
+                return []
+
+            # Stop HMS pipeline
+            self._ws.pipelines.stop(pipeline.pipeline_id)
+            # Rename old pipeline first
+            self._ws.pipelines.update(
+                pipeline.pipeline_id,
+                name=f"{pipeline.pipeline_name} [OLD]",
+                clusters=get_pipeline.spec.clusters if get_pipeline.spec.clusters else None,
+                storage=get_pipeline.spec.storage if get_pipeline.spec.storage else None,
+                continuous=get_pipeline.spec.continuous if get_pipeline.spec.continuous else None,
+                deployment=get_pipeline.spec.deployment if get_pipeline.spec.deployment else None,
+                target=get_pipeline.spec.target if get_pipeline.spec.target else None,
+                libraries=get_pipeline.spec.libraries if get_pipeline.spec.libraries else None,
+            )
+
+        # Clone pipeline
+        headers = {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json',
+        }
+        body = {
+            'catalog': self._catalog_name,
+            'clone_mode': 'MIGRATE_TO_UC',
+            'configuration': {'pipelines.migration.ignoreExplicitPath': 'true'},
+            'name': f"{pipeline.pipeline_name}",
+        }
+        res = self._ws.api_client.do(
+            'POST', f'/api/2.0/pipelines/{pipeline.pipeline_id}/clone', body=body, headers=headers
+        )
+        assert isinstance(res, dict)
```

Comment:
We cannot avoid this?

Reply:
Could you add that as a comment, so that we know to clean this up once it becomes available in the SDK